### PR TITLE
Adding subscription tier update feature docs

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Purpose
Adding documentation for Subscription Update feature. 
Issue: https://github.com/wso2/product-apim/issues/5246

### CODE REVIEW

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Update Subscription Tier
```
Comment (shaniR) [inline en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md:1]: This document should move to under manage subscriptions as this a main feature.

Comment (shaniR) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md:1]: Shall we rename this to Update API Subscription Tier

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Adding an API Subscription Update Workflow
+
```
Comment (shaniR) [inline en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md:2]: Shall we  rename the title to Adding an API Subscription Tier Update workflow

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+This section explains how to attach a custom workflow to the API subscription update operation in the API Manager. First, see [Workflow Extensions](_Managing_Workflow_Extensions_) for information on different types of workflows executors.
+
+Attaching a custom workflow to API subscription update enables you to approve the throttling tier which the consumer has requested for the existing subscription. Only admins can approve and set the new tier for the existing subscription. However, when API subscription update workflow is enabled, when the consumer tries to update the tier of an active existing subscription to an API, it initially is in the `TIER_UPDATE_PENDING` state, and consumer can still use the API with the subscription with this state which still engages the previous existing tier, using its production or sandbox keys, until their subscription update to new tier is approved. Once approval happens the new requested tier will be assigned to the existing subscription with `UNBLOCKED` status.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md:5]: Yes, this workflow only enagages when throttling tier change of a subscription request comes. 

Subscription Tier upgrade feature will provide capability to change the subscription tier of an already existing subscription without having to delete the subscription and resubscribing to the same API. In the previous WSO2 API Manager releases API Consumers doesn’t have the capability to to upgrade their API subscription to a different tier, if they want a subscription with a new tier they had to delete the current subscription and re-subscribe to that relevant API with the new tier. Therefore, with the introduction of this new subscription upgrade feature above mentioned issue has been mitigated.(This is a paragraph taken from the marketing docs which explains the usefulness of the feature)

All the other subscription related stuff like throttling and analytics works same as before. no need to mention anything specifically.

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ 
+2.  Select the relevant subscriptions and Click **EDIT** icon to enter the subscription update popup.
+
+    ![[Subscriptions update_popup](../../../../assets/img/learn/subscription-update-popup-start.png)](../../../../assets/img/learn/subscription-update-popup-start.png)
+
+3.  Select the relevant throttling tier to be updated and click **Update** to continue.
+    
+    [![Subscription_update_completed](../../../../assets/img/learn/subscription-update-completed.png)](../../../../assets/img/learn/subscription-update-completed.png)
+     
+     Note that the Throttling Tier has been updated of the existing subscription once you click **Update**.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md:21]: This sentence is to describe the image, which shows the change which has happened from the previous step.

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+## Engaging the Approval Workflow Executor in the API Manager
+
+First, enable the API subscription update workflow for **Approval Workflow Executor.**
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+1.  Sign in to API Manager Management Console ( `https://<Server Host>:9443/carbon` ) and select **Browse** under **Resources** .
+
+    ![](../../../../assets/img/learn/learn-subscription-workflow-browse.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+2.  Go to the `/_system/governance/apimgt/applicationdata/workflow-extensions.xml` resource, disable the Simple Workflow Executor and enable SubscriptionUpdateApprovalWorkflowExecutor.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            <!--SubscriptionUpdate executor="org.wso2.carbon.apimgt.impl.workflow.SubscriptionUpdateSimpleWorkflowExecutor"/-->
+            <SubscriptionUpdate executor="org.wso2.carbon.apimgt.impl.workflow.SubscriptionUpdateApprovalWorkflowExecutor"/>
+        ...
+        </WorkFlowExtensions>
+    ```
+
+    The subscription update approval workflow executor is now engaged.
+
+
+3.  Sign in to the WSO2 API Developer Portal (`https://<hostname>:<port>/devportal`) and click on the Applications and select the relevant application and go to the Subscriptions tab.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+    ![[Subscriptions update_popup](../../../../assets/img/learn/subscription-update-popup-start.png)](../../../../assets/img/learn/subscription-update-popup-start.png)
+
+3.  Select the relevant throttling tier to be updated and click **Update** to continue.
+    
+    [![Subscription_update_completed](../../../../assets/img/learn/subscription-update-completed.png)](../../../../assets/img/learn/subscription-update-completed.png)
+     
+     Note that the Throttling Tier has been updated of the existing subscription once you click **Update**. 
+     
+You can follow the steps provided in adding subscription update workflow doc if you need to configure the relevant subscription update approval process. [Adding an API Subscription Update Workflow]({{base_path}}/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

File: en/mkdocs.yml [hunk 10]
```diff
@@ -136,6 +136,8 @@ nav:
                 - Manage Subscriptions:
                       - Subscribe to an API: learn/consume-api/manage-subscription/subscribe-to-an-api.md
                       - Advanced Topics:
+                            - Update Subscription Tier: learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/mkdocs.yml]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md (hunk 1 above)
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md (hunk 2 above)
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+     ![[Subscriptions overview_tab](../../../../assets/img/learn/subscriptions-overview-tab.png)](../../../../assets/img/learn/subscriptions-overview-tab.png)
+ 
+2.  Select the relevant subscriptions and Click **EDIT** icon to enter the subscription update popup.
+
+    ![[Subscriptions update_popup](../../../../assets/img/learn/subscription-update-popup-start.png)](../../../../assets/img/learn/subscription-update-popup-start.png)
+
+3.  Select the relevant throttling tier to be updated and click **Update** to continue.
+    
+    [![Subscription_update_completed](../../../../assets/img/learn/subscription-update-completed.png)](../../../../assets/img/learn/subscription-update-completed.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    
+6.  Sign in to the Admin Portal ( `https://<Server Host>:9443/admin` ), list all the tasks for API subscription update from **Tasks** --> **Subscription Update** and click on  approve or reject to approve or reject workflow pending request.
+
+    ![](../../../../assets/img/learn/subscription-update-admin-entry.png)
+
+     After approving go back to the API Devportal Application Subscriptions tab, the subscription status will be **UNBLOCKED** and, the requested tier will also be assigned.
+     
+    ![](../../../../assets/img/learn/subscription-update-completed.png)
+
+    Now the consumer can use the existing subscription with the newly assigned throttling tier.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: Yes, immediately and it's continuation

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+     
+4.  Select the relevant subscriptions and Click **EDIT** icon to enter the subscription update popup.
+
+    <a href="{{base_path}}/assets/img/learn/subscription-update-popup-start.png" ><img src="{{base_path}}/assets/img/learn/subscription-update-popup-start.png" alt="Subscription Update Popup" title="Subscription Update Popup" /></a>
+
+5.  Select the relevant throttling tier to be updated and click **Update** to continue. After updating you will see the subscription status as **TIER_UPDATE_PENDING**.
+
+    ![](../../../../assets/img/learn/subscription-update-before-approval.png)
+    
+6.  (Optional) if the consumer need to update the requested tier to a different tier, click **EDIT** icon and select the new requested tier and click **Update** button.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: Yes

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+        ...
+        </WorkFlowExtensions>
+    ```
+
+    The subscription update approval workflow executor is now engaged.
+
+
+3.  Sign in to the WSO2 API Developer Portal (`https://<hostname>:<port>/devportal`) and click on the Applications and select the relevant application and go to the Subscriptions tab.
+    
+     ![[Subscriptions overview_tab](../../../../assets/img/learn/subscriptions-overview-tab.png)](../../../../assets/img/learn/subscriptions-overview-tab.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+     
+4.  Select the relevant subscriptions and Click **EDIT** icon to enter the subscription update popup.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+    <a href="{{base_path}}/assets/img/learn/subscription-update-popup-start.png" ><img src="{{base_path}}/assets/img/learn/subscription-update-popup-start.png" alt="Subscription Update Popup" title="Subscription Update Popup" /></a>
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+5.  Select the relevant throttling tier to be updated and click **Update** to continue. After updating you will see the subscription status as **TIER_UPDATE_PENDING**.
+
+    ![](../../../../assets/img/learn/subscription-update-before-approval.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+    
+6.  (Optional) if the consumer need to update the requested tier to a different tier, click **EDIT** icon and select the new requested tier and click **Update** button.
+    
+    ![](../../../../assets/img/learn/subscription-update-new-tier-request.png)
+    
+6.  Sign in to the Admin Portal ( `https://<Server Host>:9443/admin` ), list all the tasks for API subscription update from **Tasks** --> **Subscription Update** and click on  approve or reject to approve or reject workflow pending request.
+
+    ![](../../../../assets/img/learn/subscription-update-admin-entry.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+
+     After approving go back to the API Devportal Application Subscriptions tab, the subscription status will be **UNBLOCKED** and, the requested tier will also be assigned.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+     
+    ![](../../../../assets/img/learn/subscription-update-completed.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+
+Once you have an active approved subscription you can update the existing selected tier of the subscription. 
+You can follow the steps provided in adding subscription to an API doc to create an active subscription. [Subscribe to an API]({{base_path}}/learn/consume-api/manage-subscription/subscribe-to-an-api) 
+
+
+## Steps to Update Subscription Tier
+
+1.  Sign in to the WSO2 API Developer Portal (`https://<hostname>:<port>/devportal`) and click on the Applications and select the relevant application and go to the Subscriptions tab.
+
+     ![[Subscriptions overview_tab](../../../../assets/img/learn/subscriptions-overview-tab.png)](../../../../assets/img/learn/subscriptions-overview-tab.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+ 
+2.  Select the relevant subscriptions and Click **EDIT** icon to enter the subscription update popup.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+
+    ![[Subscriptions update_popup](../../../../assets/img/learn/subscription-update-popup-start.png)](../../../../assets/img/learn/subscription-update-popup-start.png)
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+
+3.  Select the relevant throttling tier to be updated and click **Update** to continue.
```
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (shaniR) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: This is exactly what needs to be in the doc as well. Can we add this?
change the following 
1) Consumers doesn’t have -> Consumers didn't have
2) there are 2 to's -> capability to to upgrade 
3) if they want a subscription with a new tier they had to delete the current subscription and re-subscribe to that relevant API with the new tier -> if a new subscription tier is required, the current subscription needed to be deleted, and a re-subscription with the new tier was required.

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (shaniR) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: what I am not clear is, what do you mean by "Throttling Tier has been updated of the existing subscription" , do you mean to say that the existing subscripion has been updated with the newly selected throttling tier. ?? if so, let's add that.

[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: yes, will change

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: okay, will make the change

[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/adding-an-api-subscription-update-workflow.md]: resolved

File: en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (CrowleyRajapakse) [reply en/docs/learn/consume-api/manage-subscription/advanced-topics/update-subscription-tier.md]: resolved

**Pull request merged**